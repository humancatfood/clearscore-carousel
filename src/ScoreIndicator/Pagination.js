import React from 'react';
import { css, cx } from 'emotion';



const paginationStyle = css({
  position: 'absolute',
  bottom: '10%',
  left: '20%',
  width: '60%',
  display: 'flex',
  justifyContent: 'center',
});

const paginationTickStyle = css({
  width: 10,
  height: 10,
  borderRadius: '100%',
  margin: '0 5px',
  cursor: 'pointer',
  border: '2px solid white',
  boxSizing: 'border-box',
  '&.active': {
    backgroundColor: 'white',
  }
});


const Pagination = ({ numPages, currentPageIndex, setCurrentPageIndex }) => {
  if (numPages > 1) {
    return (
      <div className={paginationStyle}>
        {
          Array(numPages).fill(undefined).map((_, index) => (
            <div
              key={index}
              onClick={() => setCurrentPageIndex(index)}
              className={cx(paginationTickStyle, {active: index === currentPageIndex})}
            ></div>
          ))
        }
      </div>
    );
  } else {
    return null;
  }
};

export default Pagination;
