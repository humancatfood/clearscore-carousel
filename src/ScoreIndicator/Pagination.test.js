import React from 'react';
import TestRenderer from 'react-test-renderer';

import Pagination, { paginationStyle, paginationTickStyle } from './Pagination';



describe('Pagination', () => {

  let element;

  it('should render', () => {

    element = TestRenderer.create(<Pagination />);
    expect(element.toJSON()).toMatchSnapshot();
    expect(element.toJSON()).toBeNull();

    element = TestRenderer.create(<Pagination numPages={0} />);
    expect(element.toJSON()).toMatchSnapshot();
    expect(element.toJSON()).toBeNull();

    element = TestRenderer.create(<Pagination numPages={5} />);
    expect(element.toJSON()).toMatchSnapshot();
    expect(element.root.findAllByType('button').length).toBe(5);

  });

  it('should be styled', () => {

    element = TestRenderer.create(<Pagination numPages={5} />);
    expect(element.toJSON().props.className).toBe(paginationStyle);
    element.root.findAllByType('button').forEach(child => {
      expect(child.props.className).toBe(paginationTickStyle);
    });

    [-5, 0, 2, 5, 50, 0.32442].forEach(indexToTest => {

      element.update(<Pagination numPages={5} currentPageIndex={indexToTest} />);
      expect(element.toJSON().props.className).toBe(paginationStyle);
      element.root.findAllByType('button').forEach((child, index) => {
        if (index === indexToTest) {
          expect(child.props.className).toBe(`${paginationTickStyle} active`);
        } else {
          expect(child.props.className).toBe(paginationTickStyle);
        }
      });

    });

  });

  it('should accept clicks', () => {

    const onClick = jest.fn();
    element = TestRenderer.create(<Pagination
      numPages={3}
      currentPageIndex={0}
      setCurrentPageIndex={onClick}
    />);

    const buttons = element.root.findAllByType('button');

    buttons.forEach((button, indexToClick) => {
      button.props.onClick();
      expect(onClick).toHaveBeenCalledWith(indexToClick);
    });

  });

});
