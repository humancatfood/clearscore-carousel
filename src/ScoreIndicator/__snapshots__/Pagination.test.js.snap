// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination should render 1`] = `null`;

exports[`Pagination should render 2`] = `null`;

exports[`Pagination should render 3`] = `
<div
  className="css-i65pe3"
>
  <button
    className="css-1mdttzm"
    onClick={[Function]}
    type="button"
  />
  <button
    className="css-1mdttzm"
    onClick={[Function]}
    type="button"
  />
  <button
    className="css-1mdttzm"
    onClick={[Function]}
    type="button"
  />
  <button
    className="css-1mdttzm"
    onClick={[Function]}
    type="button"
  />
  <button
    className="css-1mdttzm"
    onClick={[Function]}
    type="button"
  />
</div>
`;
